src/Hamiltonians/Sum.jl

```diff
@@ -0,0 +1,142 @@
+"""
+    add(A::AbstractHamiltonian, B::AbstractHamiltonian; a=1, b=1, weight=0.5) -> HamiltonianSum
+    HamiltonianSum(A::AbstractHamiltonian, B::AbstractHamiltonian; weight=0.5)
+    +(A::AbstractHamiltonian, B::AbstractHamiltonian)
+
+The sum of two [`AbstractHamiltonian`](@ref)s with coefficients, `aA + bB`. The two
+Hamiltonians must act on the same address space. The keyword argument `weight` is the
+probability of random spawns from `A`, with `1 - weight` the probability of spawning from
+`B`.
+"""
+struct HamiltonianSum{T, H1<:AbstractHamiltonian, H2<:AbstractHamiltonian} <: AbstractHamiltonian{T}
+    h1::H1
+    h2::H2
+    weight::Float64
+end
+function HamiltonianSum(h1::AbstractHamiltonian{T1}, h2::AbstractHamiltonian{T2}; weight=0.5) where {T1, T2}
+    if !(allows_address_type(h2, starting_address(h1))) || !(allows_address_type(h1, starting_address(h2)))
+        throw(ArgumentError("The Hamiltonians are not compatible."))
+    end
+    T = promote_type(T1,T2)
+    return HamiltonianSum{T, typeof(h1), typeof(h2)}(h1, h2, min(abs(weight), 1.0))
+end
+Base.:+(h1::AbstractHamiltonian, h2::AbstractHamiltonian) = HamiltonianSum(h1, h2)
+
+function Base.show(io::IO, s::HamiltonianSum)
+    print(io, "HamiltonianSum(", s.h1, ", ", s.h2, "; weight=", s.weight, ")")
+end
+
+function VectorInterface.add(h1::AbstractHamiltonian, h2::AbstractHamiltonian, a::Number, b::Number; weight=0.5)
+    return HamiltonianSum(a*h1, b*h2; weight)
+end
```

Comment:
I get an error when I try `add` without the factors:
```julia-repl
julia> add(h1, h2; weight = 0.8)
ERROR: MethodError: no method matching add(::HubbardReal1D{Float64, BoseFS{…}, 1.0, 1.0}, ::HubbardReal1DEP{Float64, BoseFS{…}, 1.0, 1.0, 3}; weight::Float64)
This method may not support any kwargs.

Closest candidates are:
  add(::AbstractHamiltonian, ::AbstractHamiltonian, ::Number, ::Number; weight)
   @ Rimu ~/git/code/Rimu/src/Hamiltonians/Sum.jl:29
  add(::Any, ::Any) got unsupported keyword argument "weight"
   @ VectorInterface ~/.julia/packages/VectorInterface/J6qCR/src/interface.jl:107
  add(::Any, ::Any, ::Number, ::Number) got unsupported keyword argument "weight"
   @ VectorInterface ~/.julia/packages/VectorInterface/J6qCR/src/fallbacks.jl:129
  ...
```
Maybe we should add additional methods?
```suggestion
function VectorInterface.add(h1::AbstractHamiltonian, h2::AbstractHamiltonian, a::Number=1, b::Number=1; weight=0.5)
    return HamiltonianSum(a*h1, b*h2; weight)
end
```

Reply:
Oh, and please add a docstring!